main.cpp

```diff
@@ -148,7 +117,17 @@ template<typename T>
 struct Numeric
 {
     using Type = Temporary<T>;
-    explicit Numeric(Type val) : value(std::make_unique<Type>(val)) {}
+    Numeric(Type&& val) : value(std::make_unique<Type>(std::forward<Type>(val))) {}
```

Comment:
This constructor argument should be `T`, not `Type`. 
You pass the primitives, not `Temporary<>` instances. 
What you have written forces an implicit conversion from the primitive into a `Temporary<>` when this constructor is called with something like `Numeric<float> f(2.5f);`
the `2.5f` is implicitly converted to an  R-Value`Temporary<float>`, which means it has no name. 
it is then forwarded along to the `Temporary<float>` **move constructor**, which is used when initializing `value` with the proper type.
- [ ] revise